### PR TITLE
Add funciton coalesce 

### LINKED ISSUE #5768: COALESCE function
**Summary**

Category: Conditional Functions

Syntax:
```
COALESCE( <expr1> , <expr2> [ , ... , <exprN> ] )
```

Example:
```
SELECT COALESCE(NULL, NULL, NULL, 'databend.com', NULL, 'snow.com');

databend.com
```

Reference:
https://docs.snowflake.com/en/sql-reference/functions/coalesce.html

### PR DESCRIPTION
**Pull request by TianLangStudio**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add funciton coalesce  WIP

## Changelog

- New Feature

## Related Issues

Fixes #5768

### CODE REVIEW

**Comment by sundy-li** [comment]
@TianLangStudio 
Refer to : https://github.com/ClickHouse/ClickHouse/blob/70b51133c11bad4afb905d53e08ccf2956c19f68/src/Functions/coalesce.cpp#L92-L98

We can transform `coalesce ` expression inside the type_checker  (in the new planner)

**Comment by TianLangStudio** [comment]
image
image

**Comment by TianLangStudio** [comment]
@sundy-li Thanks. As you know I am a newbie. I thought it didn't support this way until I saw the function NULLIF last week. I will commit later and may not have time to add tests and documents recently. Maybe you can review the function coalesce first.
It may not be the same as the reference you gave.  
Nice weekend!

File: query/src/sql/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -586,6 +588,39 @@ impl<'a> TypeChecker<'a> {
                 .await?
             }
 
+            Expr::Coalesce { span, exprs } => {
+                // Rewrite COALESCE(expr1, expr2, ...) to IF(IS_NOT_NULL(expr1), expr1, expr2)   
+                let resolve_args = exprs.iter().rev().fold(
```
Comment (sundy-li) [inline query/src/sql/planner/semantic/type_check.rs:593]: We can filter exprs to remove `Literal::Null`

Comment (TianLangStudio) [reply query/src/sql/planner/semantic/type_check.rs:593]: I'll add the filter later.

**Review by leiysky — APPROVED** [review]

**Pull request merged**